sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataWriterDAO.java

```diff
@@ -195,8 +201,23 @@ private Statement deleteStatement(TableDDL tableDDL, ByteBuffer rowKey, UUID cha
                 .setConsistencyLevel(consistencyLevel);
     }
 
+    /**
+     * This method returns a delete statement for a single block, rather than the delta as whole. In order to delete an
+     * entire delta using this method, it should be called once for each block. In cases in which there are relatively
+     * few blocks in a delta, this method can be more efficient than {@link #deleteStatement(TableDDL, ByteBuffer, UUID, ConsistencyLevel)},
+     * as the cell tombstones created by this method are more efficient than range tombstones
+     */
+    private Statement blockDeleteStatement(BlockedDeltaTableDDL tableDDL, ByteBuffer rowKey, UUID changeId, int block, ConsistencyLevel consistencyLevel) {
+        return QueryBuilder.delete()
+                .from(tableDDL.getTableMetadata())
+                .where(eq(tableDDL.getRowKeyColumnName(), rowKey))
+                .and(eq(tableDDL.getChangeIdColumnName(), changeId))
+                .and(eq(tableDDL.getBlockColumnName(), block))
+                .setConsistencyLevel(consistencyLevel);
+    }
+
     private void deleteCompactedDeltas(ByteBuffer rowKey, WriteConsistency consistency, DeltaPlacement placement,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS3776)